driver.py

```diff
@@ -22,9 +24,27 @@
 class SmartSimScalingTests:
 
     def __init__(self):
-        self.resnet_model = "./imagenet/resnet50.pt"
         self.date = str(datetime.datetime.now().strftime("%Y-%m-%d"))
 
+    def set_resnet_model(self, device="GPU", force_rebuild=False):
+        self.resnet_model = f"./imagenet/resnet50.{device}.pt"
+        if not Path(self.resnet_model).exists() or force_rebuild:
+            logger.info(f"AI Model {self.resnet_model} does not exist or rebuild was asked, it will be created")
+            save_model(device=device)
+
+        logger.info(f"Using model {self.resnet_model}")
+
+    def _check_model(self, device, force_rebuild=False):
+        if device.startswith("GPU") and (force_rebuild or not Path("./imagenet/resnet50.GPU.pt").exists()):
+            from torch.cuda import is_available
+            if not is_available():
+                message = "resnet50.GPU.pt model missing in ./imagenet directory. \n"
```

Comment:
This is - admittedly - a bit convoluted. This function `_check_model` only checks whether the GPU model is missing and, if that's the case, if CUDA is not available. It is called immediately, before we start setting up experiments, orchestrators, and so on, because we know they would fail. The checks you are suggesting are performed in another function, `set_resnet_model`, on lines 29-35

Reply:
ahhh I see